slybot/slybot/plugins/scrapely_annotations/builder.py

```diff
@@ -308,12 +309,33 @@ def apply_selector_annotations(annotations, target_page):
             elems = accepted_elements - rejected_elements
         else:
             elems = [elem._root for elem in page.css(annotation['selector'])]
-        if elems:
-            tagids = [int(e.attrib.get('data-tagid', 1e9)) for e in elems]
-            tagid = min(tagids)
-            if tagid is not None:
-                annotation['tagid'] = tagid
-                converted_annotations.append(annotation)
+        if not elems:
+            continue
+
+        tagids = [int(e.attrib.get('data-tagid', 1e9)) for e in elems]
+        tagid = min(tagids)
+        if tagid is not None:
+            annotation['tagid'] = tagid
+            converted_annotations.append(annotation)
+        if (annotation.get('repeated') and
+                not annotation.get('item_container') and
+                len(annotation.get('annotations')) == 1):
+            parent = _get_parent(elems, page)
+            field = annotation['annotations'].values()[0][0]['field']
+            container_id = '%s#parent' % annotation['id']
+            if len(parent):
+                converted_annotations.append({
+                    'item_container': True,
+                    'id': container_id,
+                    'annotations': {'#portia-content': '#dummy'},
+                    'text-content': '#portia-content',
+                    'container_id': annotation['container_id'],
+                    'field': field,
+                    'tagid': parent.attrib.get('data-tagid')
+                })
+                annotation['item_container'] = True
+                annotation['field'] = field
+                annotation['container_id'] = container_id
```

Comment:
They are meant to be mutated. This function used to just add the `tagid` key to the annotations so they would be applied to the correct elements. Now it also adds support for building a `RepeatedFieldsExtractor` by converting a repeated annotation to a container and a repeated field container.

Here's a quick overview of the 3 different types of containers:
**Container:** `item_container == True`.
**RepeatedContainer:** `item_container == True and repeated == True`.
**RepeatedFieldsContainer:** `item_container == True and repeated == True and field == True`.